src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
> What's your specific concern?

The advice given on several issues (here and the mailing list) is to avoid using instance variables in conditions when the variable is going to be used later on, since the value present on it might change (or be `nil`)

```crystal
if @something
  puts @something # this might not be the same as above line
end
```

I was wondering if that applies to this case: when the variable is used internally and on `case` condition.